exoscale/datasource_exoscale_domain_record.go

```diff
@@ -0,0 +1,196 @@
+package exoscale
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/exoscale/egoscale"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func datasourceDomainRecord() *schema.Resource {
+	return &schema.Resource{
+
+		Read: datasourceDomainRecordRead,
+
+		Schema: map[string]*schema.Schema{
+			"domain": {
+				Type:        schema.TypeString,
+				Description: "Domain of the Record",
+				Required:    true,
+			},
+			"filter": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:          schema.TypeInt,
+							Optional:      true,
+							ConflictsWith: []string{"filter.0.name", "filter.0.record_type", "filter.0.content"},
+						},
+						"name": {
+							Type:          schema.TypeString,
+							Optional:      true,
+							ConflictsWith: []string{"filter.0.id", "filter.0.record_type", "filter.0.content"},
+						},
+						"record_type": {
+							Type:          schema.TypeString,
+							Optional:      true,
+							ConflictsWith: []string{"filter.0.id", "filter.0.content"},
+						},
+						"content": {
+							Type:          schema.TypeString,
+							Optional:      true,
+							ConflictsWith: []string{"filter.0.id", "filter.0.name", "filter.0.record_type"},
+						},
+					},
+				},
+			},
+			"records": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:        schema.TypeInt,
+							Description: "ID of the Record",
+							Optional:    true,
+						},
+						"domain": {
+							Type:        schema.TypeString,
+							Description: "Domain of the Record",
+							Optional:    true,
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Description: "Name of the Record",
+							Optional:    true,
+						},
+						"content": {
+							Type:        schema.TypeString,
+							Description: "Content of the Record",
+							Optional:    true,
+						},
+						"create_at": {
+							Type:        schema.TypeString,
+							Description: "Creation of the Record",
+							Optional:    true,
+						},
+						"update_at": {
+							Type:        schema.TypeString,
+							Description: "Last update of the Record",
+							Optional:    true,
+						},
```

Comment:
I think we should mirror the `exoscale_domain_record` attributes for consistency here, and not introduce date-related attributes – unless you have a very good use case for those with Terraform...